File: src/app/shared/services/ticket.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Ticket, TicketRequest, TicketMessage, TicketMessageRequest } from '../models/ticket.model';

@Injectable({ providedIn: 'root' })
export class TicketService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/tickets`;

  getTickets(): Observable<Ticket[]> {
    return this.http.get<Ticket[]>(this.apiUrl);
  }

  createTicket(request: TicketRequest): Observable<Ticket> {
    return this.http.post<Ticket>(this.apiUrl, request);
  }

  getMessages(ticketId: string): Observable<TicketMessage[]> {
    return this.http.get<TicketMessage[]>(`${this.apiUrl}/${ticketId}/messages`);
  }

  sendMessage(ticketId: string, request: TicketMessageRequest): Observable<TicketMessage> {
    return this.http.post<TicketMessage>(`${this.apiUrl}/${ticketId}/messages`, request);
  }
}

File: backend/src/main/java/com/zenpay/application/service/TicketService.java
package com.zenpay.application.service;

import com.zenpay.application.dto.*;
import com.zenpay.domain.model.*;
import com.zenpay.domain.repository.TicketMessageRepository;
import com.zenpay.domain.repository.TicketRepository;
import com.zenpay.infrastructure.exception.BusinessException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class TicketService {

    private final TicketRepository ticketRepository;
    private final TicketMessageRepository ticketMessageRepository;

    public List<TicketResponse> getTickets(UUID userId) {
        return ticketRepository.findByUserId(userId).stream()
                .map(this::toTicketResponse)
                .collect(Collectors.toList());
    }

    @Transactional
    public TicketResponse createTicket(UUID userId, TicketRequest request) {
        Ticket ticket = Ticket.builder()
                .user(User.builder().id(userId).build())
                .subject(request.subject())
                .description(request.description())
                .priority(request.priority() != null ? request.priority() : TicketPriority.MEDIUM)
                .category(request.category())
                .build();

        ticket = ticketRepository.save(ticket);
        return toTicketResponse(ticket);
    }

    public List<TicketMessageResponse> getMessages(UUID ticketId) {
        return ticketMessageRepository.findByTicketIdOrderByCreatedAtAsc(ticketId).stream()
                .map(m -> new TicketMessageResponse(
                        m.getId(), m.getMessage(), m.getSender(), m.getCreatedAt()))
                .collect(Collectors.toList());
    }

    @Transactional
    public TicketMessageResponse sendMessage(UUID ticketId, TicketMessageRequest request, String sender) {
        Ticket ticket = ticketRepository.findById(ticketId)
                .orElseThrow(() -> new BusinessException("TICKET_NOT_FOUND", "Ticket not found"));

        TicketMessage message = TicketMessage.builder()
                .ticket(ticket)
                .sender(sender)
                .message(request.message())
                .build();

        message = ticketMessageRepository.save(message);
        return new TicketMessageResponse(
                message.getId(), message.getMessage(), message.getSender(), message.getCreatedAt());
    }

    private TicketResponse toTicketResponse(Ticket ticket) {
        return new TicketResponse(
                ticket.getId(), ticket.getSubject(), ticket.getDescription(),
                ticket.getStatus(), ticket.getPriority(), ticket.getCategory(),
                ticket.getCreatedAt(), ticket.getUpdatedAt());
    }
}
